app.rb

```diff
@@ -266,6 +266,7 @@ def validate!
     end
 
     destroy do
+      resource.cleanup
```

Comment:
Can we move this to `EventProcessor.cancel_job`.  We're eventually going to want to handle job cancellations differently, for the different types of jobs.  Having all of the logic in a single location is going to make that easier to refactor.

Reply:
Thanks.